BMM.Core/Models/POs/ContinueListening/ContinueListeningTilePO.cs

```diff
@@ -71,10 +77,18 @@ public bool IsCurrentlyPlaying
             set => SetProperty(ref _isCurrentlyPlaying, value);
         }
         
-        public void RefreshState()
+        public bool IsDownloaded
+        {
+            get => _isDownloaded;
+            set => SetProperty(ref _isDownloaded, value);
+        }
+        
+        public Task RefreshState()
         {
             IsCurrentlySelected = _mediaPlayer.CurrentTrack != null && _mediaPlayer.CurrentTrack.Id.Equals(ContinueListeningTile.Track.Id);
             IsCurrentlyPlaying = IsCurrentlySelected && _mediaPlayer.IsPlaying;
+            IsDownloaded = _storageManager.SelectedStorage.IsDownloaded(ContinueListeningTile.Track);
```

Comment:
should this be awaited?
If we actually check the filesystem I'd expect this to be async.

Reply:
checked it. it's fine